### PR TITLE
Drop remaining jobs, update migration report doc, enforce not adding any new default cluster jobs

### PR DESCRIPTION
**Pull request by BenTheElder**
I will be scaling the old build cluster behind `cluster: default` down to zero nodes shortly, then we can look at dropping it from prow's KUBECONFIGs.

This fixes https://github.com/kubernetes/test-infra/issues/33226

### CODE REVIEW

File: config/tests/jobs/jobs_test.go [hunk 1]
```diff
@@ -293,160 +293,34 @@ func TestCommunityJobs(t *testing.T) {
...
 	knownBadJobs := sets.NewString(
-		"capz-azure-disk-master",
-		"capz-azure-disk-vmss-master",
-		"capz-azure-file-master",
-		"capz-azure-file-vmss-master",
-		"capz-conformance-dual-stack-master",
-		"capz-conformance-ipv6-master",
-		"capz-conformance-master",
 		"ci-test-infra-gencred-refresh-kubeconfig",
 		"ci-test-infra-rotate-legacy-default-build-sa-json-key",
```
Comment (BenTheElder) [inline config/tests/jobs/jobs_test.go:301]: we should be able to drop this one next, but I want to have the cluster actually not running jobs first and then confirm

**Review by upodroid — APPROVED** [review]
LGTM 🎉

**Comment by BenTheElder** [comment]
fixed yamllint: no EOL at end of file in two files, testgrid config referencing generated dashboards that are now emtpy

File: config/tests/jobs/jobs_test.go (hunk 1 above)
Comment (BenTheElder) [reply config/tests/jobs/jobs_test.go:301]: talked to cole about this, we can clean it up.

Comment (BenTheElder) [reply config/tests/jobs/jobs_test.go:301]: https://github.com/kubernetes/test-infra/pull/33274

**Pull request merged**